gateway/ln-gateway/src/gateway_module_ng/receive_sm.rs

```diff
@@ -0,0 +1,243 @@
+use std::collections::BTreeMap;
+use std::future::pending;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::{anyhow, bail};
+use fedimint_client::sm::{ClientSMDatabaseTransaction, State, StateTransition};
+use fedimint_client::transaction::ClientInput;
+use fedimint_client::DynGlobalClientContext;
+use fedimint_core::api::{deserialize_outcome, FederationApiExt, SerdeOutputOutcome};
+use fedimint_core::core::{Decoder, OperationId};
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::endpoint_constants::AWAIT_OUTPUT_OUTCOME_ENDPOINT;
+use fedimint_core::module::ApiRequestErased;
+use fedimint_core::query::FilterMapThreshold;
+use fedimint_core::task::sleep;
+use fedimint_core::{NumPeersExt, OutPoint, PeerId, TransactionId};
+use fedimint_ln_client_ng::LightningClientStateMachines;
+use fedimint_ln_common_ng::contracts::IncomingContract;
+use fedimint_ln_common_ng::{LightningInput, LightningOutputOutcome, Witness};
+use secp256k1::KeyPair;
+use tpe::{aggregate_decryption_shares, AggregatePublicKey, DecryptionKeyShare, PublicKeyShare};
+use tracing::trace;
+
+use crate::gateway_module_ng::GatewayClientContextNG;
+
+const RETRY_DELAY: Duration = Duration::from_secs(1);
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+pub struct ReceiveStateMachine {
+    pub operation_id: OperationId,
+    pub contract: IncomingContract,
+    pub out_point: OutPoint,
+    pub state: ReceiveSMState,
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+pub enum ReceiveSMState {
```

Comment:
These state machines should be much simpler than before, but can we still add the mermaid diagram? I found those helpful